docs/admin/cluster-components.md

```diff
@@ -101,14 +126,38 @@ to create these pods and services does run on the master VM. See:
 
 Addon objects are created in the "kube-system" namespace.
 
-Example addons are:
-* [DNS](http://releases.k8s.io/HEAD/cluster/addons/dns/) provides cluster local DNS.
-* [kube-ui](http://releases.k8s.io/HEAD/cluster/addons/kube-ui/) provides a graphical UI for the
-  cluster.
-* [fluentd-elasticsearch](http://releases.k8s.io/HEAD/cluster/addons/fluentd-elasticsearch/) provides
-  log storage. Also see the [gcp version](http://releases.k8s.io/HEAD/cluster/addons/fluentd-gcp/).
-* [cluster-monitoring](http://releases.k8s.io/HEAD/cluster/addons/cluster-monitoring/) provides
-  monitoring for the cluster.
+#### DNS
+
+While the other addons are not strictly required, all Kubernetes
+clusters should have [cluster DNS], as many examples rely on it.
+
+Cluster DNS is a DNS server, in addition to the other DNS server(s) in your
+environment, which serves DNS records for Kubernetes services.
+
+Containers started by Kubernetes automatically include this DNS server
+in their DNS searches.
+
+#### User interface
+
+The kube-ui provides a read-only overview of the cluster state.  Access
+[the UI using kubectl proxy](../user-guide/connecting-to-applications-proxy.md#connecting-to-the-kube-ui-service-from-your-local-workstation)
+
+#### Container Resource Monitoring
+
+[Container Resource Monitoring](../user-guide/monitoring.md) records generic time-series metrics
+about containers in a central database, and provides a UI for browsing that data.
+
+#### Cluster-level Logging
+
+[Container Logging](../user-guide/monitoring.md) saves container logs
+to a central log store with search/browsing interface.  There are two
+implementations:
+
+* [Cluster level logging to Google Cloud Logging](
```

Comment:
I agree hyphenated is better.
Only first word of hypenated pair is capitalized in a heading:
http://www.accu-assist.com/grammar-tips-archive/GrammarTip_capitalization-titles-headings-hyphenated-words.htm